src/cat-data-service/src/settings.rs

```diff
@@ -26,3 +58,288 @@ pub struct Settings {
+    fn invalid_hostname(hostname: &str) -> String {
+        error!("Invalid hostname for API: {}", hostname);
+        String::new()
+    }
+
+    let configured_hosts: Vec<String> = hosts
+        .split(',')
+        .map(|s| {
+            let url = Url::parse(s.trim());
+            match url {
+                Ok(url) => {
+                    // Get the scheme, and if its empty, use http
+                    let scheme = url.scheme();
+
+                    let port = url.port();
+
+                    // Rebuild the scheme + hostname
+                    match url.host() {
+                        Some(host) => {
+                            let host = host.to_string();
+                            if host.is_empty() {
+                                invalid_hostname(s)
+                            } else {
+                                match port {
+                                    Some(port) => {
+                                        format! {"{scheme}://{host}:{port}"}
+                                        //scheme.to_owned() + "://" + &host + ":" + &port.to_string()
+                                    }
+                                    None => {
+                                        format! {"{scheme}://{host}"}
+                                    }
+                                }
+                            }
+                        }
+                        None => invalid_hostname(s),
+                    }
+                }
+                Err(_) => invalid_hostname(s),
+            }
+        })
+        .filter(|s| !s.is_empty())
+        .collect();
+
+    // If there are no hostnames, just use the address of the service.
+    if configured_hosts.is_empty() {
+        // If the Socket Address is the "catchall" address, then use localhost.
+        if match addr.ip() {
+            IpAddr::V4(ipv4) => ipv4.is_unspecified(),
+            IpAddr::V6(ipv6) => ipv6.is_unspecified(),
+        } {
+            let port = addr.port();
+            vec![format! {"http://localhost:{port}"}]
+        } else {
+            vec![format! {"http://{addr}"}]
+        }
+    } else {
+        configured_hosts
+    }
+}
+
+/// Get a list of all hostnames to serve the API on.
+///
+/// Used by the `OpenAPI` Documentation to point to the correct backend.
+/// Take a list of [scheme://] + hostnames from the env var and turns it into
+/// a lits of strings.
+///
+/// Hostnames are taken from the `API_HOSTNAMES` environment variable.
+/// If that is not set, `addr` is used.
+pub(crate) fn get_api_hostnames(addr: &SocketAddr) -> Vec<String> {
+    string_to_api_hostnames(addr, API_HOSTNAMES.as_str())
+}
+
+// Jorm cleanup timeout is only used if feature is enabled.
+#[cfg(feature = "jorm-mock")]
+lazy_static! {
+    /// The jorm mock timeout, only used if feature is enabled.
+    pub(crate) static ref JORM_CLEANUP_TIMEOUT: Duration = {
+        dotenv().ok();
```

Comment:
Why do we need that, it makes no effect.
Or I am missing something

Reply:
I am not sure what you mean by "no effect"?
JORM_CLEANUP_TIMEOUT is read once and converted once.  any other time its needed the pre-read and converted result is just used?

Reply:
Ohh, sorry just read about dotenv() function